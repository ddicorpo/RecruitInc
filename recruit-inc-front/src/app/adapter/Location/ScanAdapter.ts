/**
 * Take JSON return all our current location we are scanning.
 */
export class ScanAdapter {
  public adapt(rawJSON): string[] {
    let locationScanning: string[] = [];
    for (let index = 0; index < rawJSON.length; index++) {
      const loc: string = rawJSON[index].location;
      locationScanning.push(loc);
    }
    return locationScanning;
  }
}
