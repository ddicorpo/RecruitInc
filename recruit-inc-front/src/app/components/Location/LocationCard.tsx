import React from 'react';
import { LocationCardProps } from '../../props/LocationCardProps';

export class LocationCard extends React.Component<LocationCardProps, {}> {
  constructor(props: LocationCardProps) {
    super(props);
  }

  render() {
    return <p>location</p>;
  }
}
