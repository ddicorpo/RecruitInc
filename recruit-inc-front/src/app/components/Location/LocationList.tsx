import React from 'react';
import { LocationListProps } from '../../props/LocationListProps';

export class LocationList extends React.Component<LocationListProps, {}> {
  constructor(props: LocationListProps) {
    super(props);
  }

  render() {
    return <p> Location List </p>;
  }
}
