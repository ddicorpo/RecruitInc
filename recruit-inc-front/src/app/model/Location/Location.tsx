export class Location {
  name: string;
  numberOfUsers: number;
  scannedUsers: number;

  constructor(name: string, numberOfUsers: number, scannedUsers: number) {
    this.name = name;
    this.numberOfUsers = numberOfUsers;
    this.scannedUsers = scannedUsers;
  }
}
