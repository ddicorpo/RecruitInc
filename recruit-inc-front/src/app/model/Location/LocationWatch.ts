export class LocationWatch {
  name: string;
  numberOfUsers: number;
  scannedUsers: number;

  constructor(name: string, numberOfUsers: number, scannedUsers: number) {
    this.name = name;
    this.numberOfUsers = numberOfUsers;
    this.scannedUsers = scannedUsers;
  }
}
