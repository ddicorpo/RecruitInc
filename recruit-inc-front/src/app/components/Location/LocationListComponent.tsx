import React from 'react';
import { LocationListProps } from '../../props/LocationListProps';
import { LocationCard } from './LocationCard';

export class LocationListComponent extends React.Component<
  LocationListProps,
  {}
> {
  constructor(props: LocationListProps) {
    super(props);
  }

  createUniqueId(locationName: string): string {
    const randomNumber: number = Math.floor(Math.random() * 100 + 1);
    return locationName.replace(/ /g, '') + randomNumber;
  }

  render() {
    return (
      <div className="row">
        {this.props.Locations.LocationList.map(l => (
          <LocationCard key={l.name} locationWatchData={l} />
        ))}
      </div>
    );
  }
}
