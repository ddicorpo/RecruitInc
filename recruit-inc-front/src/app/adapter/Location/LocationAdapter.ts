import { LocationWatch } from '../../model/Location/LocationWatch';

/**
 * Take JSON return all our current location we are scanning.
 */
export class LocationAdapter {
  public adapt(
    locationName: string,
    allUserJSON: any[],
    unscannedUserJSON: any[]
  ): LocationWatch {
    let numberOfUser: number = 0;
    let numberOfScannedUser: number = 0;
    numberOfScannedUser = allUserJSON.length - unscannedUserJSON.length;
    numberOfUser = allUserJSON.length;

    return new LocationWatch(locationName, numberOfUser, numberOfScannedUser);
  }
}
