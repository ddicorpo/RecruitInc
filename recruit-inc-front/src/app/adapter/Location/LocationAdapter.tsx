import { Location } from 'src/app/model/Location/Location';

/**
 * Take JSON return all our current location we are scanning.
 */
export class LocationAdapter {
  public adapt(locationName: string, allUserJSON, unscannedUserJSON): Location {
    let numberOfUser: number = 0;
    let numberOfScannedUser: number = 0;

    numberOfScannedUser = allUserJSON.length - unscannedUserJSON.lenght;
    numberOfUser = allUserJSON.lenght;

    return new Location(locationName, numberOfUser, numberOfScannedUser);
  }
}
