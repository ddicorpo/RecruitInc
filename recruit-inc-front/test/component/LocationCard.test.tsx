import { LocationWatch } from '../../src/app/model/Location/LocationWatch';
import { LocationCard } from '../../src/app/components/Location/LocationCard';
import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';

describe('LocationCard Component Testing', () => {
  it('Testing Component with data', () => {
    const fakeWatchLocation: LocationWatch = new LocationWatch('bob', 200, 30);
    const props = {
      locationWatchData: fakeWatchLocation,
    };

    const wrapper = shallow(<LocationCard {...props} />);
    const itShallContains: boolean = wrapper.contains(fakeWatchLocation.name);
    expect(itShallContains).toBe(true);
  });

  it('Testing Component full list', () => {
    const fakeWatchLocation: LocationWatch = new LocationWatch('bob', 200, 200);
    const props = {
      locationWatchData: fakeWatchLocation,
    };

    const wrapper = shallow(<LocationCard {...props} />);
    const itShallContains: boolean = wrapper.contains('Finished');
    expect(itShallContains).toBe(true);
  });

  it('Testing Component non-full list', () => {
    const fakeWatchLocation: LocationWatch = new LocationWatch('bob', 200, 20);
    const props = {
      locationWatchData: fakeWatchLocation,
    };

    const wrapper = shallow(<LocationCard {...props} />);
    const itShallContains: boolean = wrapper.contains('Running');
    expect(itShallContains).toBe(true);
  });

  it('render correctly the LocationCard component', () => {
    const fakeWatchLocation: LocationWatch = new LocationWatch('rena', 270, 20);
    const props = {
      locationWatchData: fakeWatchLocation,
    };
    const wrapper = renderer.create(<LocationCard {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
