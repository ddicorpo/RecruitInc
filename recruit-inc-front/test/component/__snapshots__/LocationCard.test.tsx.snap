// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCard Component Testing render correctly the LocationCard component 1`] = `
<div
  className="col-xl-6 col-lg-6 col-md-12 col-sm-12"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <h5
        className="super-title card-title "
      >
        rena
      </h5>
      <br />
      <ul
        className="list-group list-group-flush"
      >
        <li
          className="list-group-item"
        >
          <b>
            Number of user:
          </b>
          <span>
             
          </span>
          270
        </li>
        <li
          className="list-group-item"
        >
          <b>
            Scanned users:
          </b>
          <span>
             
          </span>
          20
        </li>
        <li
          className="list-group-item"
        >
          <b>
            Status:
          </b>
          <span>
             
          </span>
          <span>
            Running
          </span>
        </li>
      </ul>
      <div
        className="super-icon"
      >
        <i
          className="fa fa-sync fa-3x fa-spin"
        />
      </div>
    </div>
  </div>
</div>
`;
